airflow/jobs/scheduler_job.py

```diff
@@ -402,6 +402,14 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                     # Many dags don't have a task_concurrency, so where we can avoid loading the full
                     # serialized DAG the better.
                     serialized_dag = self.dagbag.get_dag(dag_id, session=session)
+                    # If the dag is missing, continue to the next task.
+                    if not serialized_dag:
+                        self.log.error(
+                            "DAG '%s' for taskinstance %s not found in serialized_dag table",
+                            dag_id,
+                            task_instance,
+                        )
+                        continue
```

Comment:
If we can't find them, shouldn't these be set to failed or something, otherwise these TIs will always come up each time around the loop

Reply:
From my testing, the scheduler continues to run the task instances even though the dag is no more.
I think we should not fail it but allow it to finish since it's not stuck or raising exception

Reply:
> allow it to finish

If the code is not there anymore, how can the task instance execute? What am I missing here 🤔

Reply:
Sorry for the delay, wanted to properly reproduce it before responding.
Here's a reproduction step:

Run this dag:
```
from airflow import DAG
from datetime import datetime

dag = DAG(
    "airflow_bug",
    schedule_interval="0 1 * * *",
    start_date=datetime(2021, 1, 1),
    max_active_runs=1,
    concurrency=1,
)

for i in range(100):
    @dag.task(task_id=f'mytasrk{i}')
    def sleeping():
        import time
        time.sleep(60)
    sleeping()
```
Once you unpause and the task start running, remove the file from the Dag folder.
Watch the scheduler logs, after some time it'll start crashing and won't recover(until the whole tasks would start failing, I think).
```
[2021-12-24 05:55:22,877] {scheduler_job.py:623} ERROR - Executor reports task instance <TaskInstance: airflow_bug.mytasrk97 scheduled__2021-01-01T01:00:00+00:00 [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally?
[2021-12-24 05:55:22,881] {scheduler_job.py:630} ERROR - Marking task instance <TaskInstance: airflow_bug.mytasrk97 scheduled__2021-01-01T01:00:00+00:00 [queued]> as failed
Traceback (most recent call last):
  File "/opt/airflow/airflow/jobs/scheduler_job.py", line 628, in _process_executor_events
    task = dag.get_task(ti.task_id)
AttributeError: 'NoneType' object has no attribute 'get_task'
```
cc: @kaxil

Reply:
This error/reproduction step is not quite right, but the same idea can trigger this behaviour -- if the dag is deleted at the "right" time, this bit of the scheduler will fail.

I think in that case though we should fail the task instances as the DAG doesn't exist anymore, and as TP said, it can't run successfully.

Reply:
Yeah, if the DAG is missing entirely, we should stop the entire run from continuing because nothing afterwards would run. If I understand correctly, skipping the task instance (as this PR currently implements) means the ti would stay queued and unnecessarily be tried again (and again…), which seems suboptimal.